### PR TITLE
Feat/479627  submit completed form2

### PR DESCRIPTION
**Pull request by whitewaterdesign**

### CODE REVIEW

File: app/lib/validation-helpers.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const validNewMicrochip = /^\d+$/
+
+const FIFTEEN = 15
+
+const validateMicrochip = (value, helpers, compareOrig = false) => {
+  let elemName = helpers.state.path[0]
+
+  // Compare new value against original to determine if already pre-populated in the DB
```
Comment (jbarnsley10) [inline app/lib/validation-helpers.js:8]: We probably don;t need lines 8 - 17 as this was to handle old format microchips. As the Police are entering new format ones, then should never need it. I suppose no harm in leaving in though, as long as it's covered by a test

File: package.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
     "date-fns-tz": "^2.0.1",
     "form-data": "^4.0.0",
     "friendly-challenge": "^0.9.13",
-    "govuk-frontend": "^5.0.0",
+    "govuk-frontend": "^5.5.0",
```
Comment (jbarnsley10) [inline package.json:42]: Do we know what changed in this version? Will it mean our pages don;t quite match the prototypes anymore?

File: app/routes/cdo/manage/tasks/generic-task-helper.js [hunk 3]
```diff
@@ -11,7 +13,7 @@ const taskList = [
   { name: tasks.applicationFeePaid, key: 'record-application-fee-payment', canSubmit: false, label: 'Application fee', apiKey: 'recordApplicationFee', stateKey: 'applicationFeePaid' },
   { name: tasks.form2Sent, key: 'send-form2', canSubmit: false, label: 'Form two', apiKey: 'sendForm2', stateKey: 'form2Sent' },
   { name: tasks.verificationDateRecorded, Model: ViewModelRecordVerificationDates, validation: validateVerificationDates, key: 'submit-form-two', canSubmit: true, label: 'Form 2 confirming dog<br>microchipped and neutered', apiKey: 'submitFormTwo', stateKey: 'verificationDateRecorded' },
-  { name: tasks.microchipDeadlineRecorded, key: 'record-microchip-deadline', canSubmit: false, label: 'When will the dog be fit to be microchipped?', apiKey: 'verifyDates', stateKey: 'verificationDateRecorded' },
+  { name: tasks.microchipDeadlineRecorded, Model: ViewModelRecordMicrochipDeadline, validation: validateMicrochipDeadlineDates, key: 'record-microchip-deadline', canSubmit: true, label: 'When will the dog be fit to be microchipped?', apiKey: 'submitFormTwo', stateKey: 'verificationDateRecorded' },
```
Comment (jbarnsley10) [inline app/routes/cdo/manage/tasks/generic-task-helper.js:16]: Is it correct that both lines 15 and 16 use the same api key of 'submitFormTwo'? Probably yes. I guess line15 api-key isn't used if you end up using the second page of line 16

File: app/views/cdo/manage/tasks/record-microchip-deadline.njk [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/date-input/macro.njk" import govukDateInput %}
+{% from "govuk/components/details/macro.njk" import govukDetails %}
+
+{% extends '_layout.njk' %}
+
+{% block pageTitle %}When will the dog be fit to be microchipped? - {{ serviceName }}{% endblock %}
+
+{% block beforeContent %}
+  <div class="govuk-grid-row">
```
Comment (jbarnsley10) [inline app/views/cdo/manage/tasks/record-microchip-deadline.njk:15]: Do you need a 'super()' in here to retain the Beta banner?

File: app/views/cdo/manage/tasks/record-microchip-deadline.njk [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        </legend>
+
+        {{ govukDateInput(model.microchipDeadline) }}
+
+        <input type="hidden" name="taskName" id="taskName" value="{{ model.taskName }}" />
+        {% if model.hidden.dogNotFitForMicrochip %}
+        <input type="hidden" name="dogNotFitForMicrochip" id="dogNotFitForMicrochip" value="" />
+        {% endif %}
+        {% if model.hidden.dogNotNeutered %}
+        <input type="hidden" name="dogNotNeutered" id="dogNotNeutered" value="" />
```
Comment (jbarnsley10) [inline app/views/cdo/manage/tasks/record-microchip-deadline.njk:53]: Is the value correct on this line?

**Review by jbarnsley10 — CHANGES_REQUESTED** [review]
All looks good to me. A few comments. Definitely one change ('super()') but not sure if any other changes needed - I was just asking questions.

File: app/views/cdo/manage/tasks/record-microchip-deadline.njk (hunk 4 above)
Comment (whitewaterdesign) [reply app/views/cdo/manage/tasks/record-microchip-deadline.njk:15]: good catch, did it with the other ones, but forgot for this

File: app/routes/cdo/manage/tasks/generic-task-helper.js (hunk 3 above)
Comment (whitewaterdesign) [reply app/routes/cdo/manage/tasks/generic-task-helper.js:16]: Yes...both need the same key, if there is not a redirect to microchip deadline, then the api is submitted on that action, then the next page also needs to call the api.

File: app/lib/validation-helpers.js (hunk 1 above)
Comment (whitewaterdesign) [reply app/lib/validation-helpers.js:8]: there is a test for this.  we can remove later as yes, guess not needed, but want to reduce work as we have 1 hr.

**Review by jbarnsley10 — APPROVED** [review]
Nice

**Pull request merged**